src/kirin/rewrite/peephole.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+
+from kirin import ir
+from kirin.dialects import py
+from kirin.rewrite.abc import RewriteRule, RewriteResult
+
+
+@dataclass
+class PeepholeOptimize(RewriteRule):
+
+    def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
+
+        if isinstance(node, py.binop.Add):
+            #   add(%a, %a) -> mul(2, %a)
+            if node.lhs == node.rhs:
```

Comment:
This is fine but each ssa value is a unique object so you really should use `is` to make this more clear. 
```suggestion
            if node.lhs is node.rhs:
```